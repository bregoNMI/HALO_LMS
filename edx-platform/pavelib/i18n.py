"""
Internationalization tasks
"""

import subprocess
import sys

from paver.easy import cmdopts, needs, sh, task

from .utils.envs import Env
from .utils.timer import timed

try:
    from pygments.console import colorize
except ImportError:
    colorize = lambda color, text: text

DEFAULT_SETTINGS = Env.DEVSTACK_SETTINGS


@task
@needs(
    "pavelib.prereqs.install_prereqs",
    "pavelib.i18n.i18n_validate_gettext",
)
@cmdopts([
    ("verbose", "v", "Sets 'verbose' to True"),
])
@timed
def i18n_extract(options):
    """
    Extract localizable strings from sources
    """
    verbose = getattr(options, "verbose", None)
    cmd = "i18n_tool extract"

    if verbose:
        cmd += " -v"

    sh(cmd)


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_generate():
    """
    Compile localizable strings from sources, extracting strings first.
    """
    sh("i18n_tool generate")


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_generate_strict():
    """
    Compile localizable strings from sources, extracting strings first.
    Complains if files are missing.
    """
    sh("i18n_tool generate --strict")


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_dummy():
    """
    Simulate international translation by generating dummy strings
    corresponding to source strings.
    """
    sh("i18n_tool dummy")
    # Need to then compile the new dummy strings
    sh("i18n_tool generate")


@task
@timed
def i18n_validate_gettext():
    """
    Make sure GNU gettext utilities are available
    """

    returncode = subprocess.call(['which', 'xgettext'])

    if returncode != 0:
        msg = colorize(
            'red',
            "Cannot locate GNU gettext utilities, which are "
            "required by django for internationalization.\n (see "
            "https://docs.djangoproject.com/en/dev/topics/i18n/"
            "translation/#message-files)\nTry downloading them from "
            "http://www.gnu.org/software/gettext/ \n"
        )

        sys.stderr.write(msg)
        sys.exit(1)


@task
@timed
def i18n_clean():
    """
    Clean the i18n directory of artifacts
    """
    sh('git clean -fdX conf/locale')
