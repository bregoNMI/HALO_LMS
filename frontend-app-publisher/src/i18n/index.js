// Placeholder messages index to be overridden by `make pull_translations`

import { messages as paragonMessages } from '@edx/paragon';
import { messages as footerMessages } from '@edx/frontend-component-footer';

export default [
  footerMessages,
  paragonMessages,
];
