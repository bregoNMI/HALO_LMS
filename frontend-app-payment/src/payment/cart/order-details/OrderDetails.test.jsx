import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';

import ConnectedOrderDetails from './OrderDetails';
import {
  SEAT_PRODUCT_TYPE,
  COURSE_ENTITLEMENT_PRODUCT_TYPE,
  ENROLLMENT_CODE_PRODUCT_TYPE,
  VERIFIED_CERTIFICATE_TYPE,
  CREDIT_CERTIFICATE_TYPE,
} from './data/constants';

jest.mock('@edx/frontend-platform/logging', () => ({
  logError: jest.fn(),
}));

const mockStore = configureMockStore();
const appContextValue = {
  authenticatedUser: {
    email: 'staff@example.com',
  },
};

describe('OrderDetails', () => {
  let state;

  beforeEach(() => {
    state = {
      userAccount: { email: 'staff@example.com' },
      payment: {
        basket: {
          loaded: false,
          loading: false,
          products: [],
        },
      },
    };
  });

  it('should render nothing by default', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    expect(tree.children.length).toBe(0);
  });

  it('should render if given a store path', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails storePath={['payment', 'basket']} />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    expect(tree.children.length).toBe(0);
  });

  it('should render nothing if given an unknown product type', () => {
    state.payment.basket.products.push({
      productType: 'something else',
      certificateType: 'verified',
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    expect(tree.children.length).toBe(0);
  });

  it('should render course entitlement product message', () => {
    state.payment.basket.products.push({
      productType: COURSE_ENTITLEMENT_PRODUCT_TYPE,
      certificateType: 'verified', // Doesn't matter
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    // This snapshot should be the message for the course entitlement product - a single sentence.
    expect(tree).toMatchSnapshot();
  });

  it('should render enrollment code product message', () => {
    state.payment.basket.products.push({
      productType: ENROLLMENT_CODE_PRODUCT_TYPE,
      certificateType: VERIFIED_CERTIFICATE_TYPE, // Doesn't matter
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <AppContext.Provider value={appContextValue}>
          <Provider
            store={mockStore(state)}
          >
            <ConnectedOrderDetails />
          </Provider>
        </AppContext.Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    // This snapshot should be a big list of bullet points and include the user's email at the end.
    expect(tree).toMatchSnapshot();
  });

  it('should render verified seat message', () => {
    state.payment.basket.products.push({
      productType: SEAT_PRODUCT_TYPE,
      certificateType: VERIFIED_CERTIFICATE_TYPE,
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    // This snapshot should be the message for the verified seat product - a single sentence.
    expect(tree).toMatchSnapshot();
  });

  it('should render credit seat message', () => {
    state.payment.basket.products.push({
      productType: SEAT_PRODUCT_TYPE,
      certificateType: CREDIT_CERTIFICATE_TYPE,
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    // This snapshot should be the message for the credit seat product - a single sentence.
    expect(tree).toMatchSnapshot();
  });

  it('should render seat message', () => {
    state.payment.basket.products.push({
      productType: SEAT_PRODUCT_TYPE,
      certificateType: 'something else', // needs to not be credit or verified.
    });
    state.payment.basket.loaded = true;
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <ConnectedOrderDetails />
        </Provider>
      </IntlProvider>
    );
    const { container: tree } = render(component);
    // This snapshot should be the message for a generic seat product - a single sentence.
    expect(tree).toMatchSnapshot();
  });
});
