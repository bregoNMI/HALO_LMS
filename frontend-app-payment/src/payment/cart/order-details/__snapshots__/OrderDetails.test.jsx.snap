// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails should render course entitlement product message 1`] = `
<div>
  <div
    class="basket-section"
  >
    <h5
      aria-level="2"
    >
      Order Details
    </h5>
    <p>
      After you complete your order you will be able to select course dates from your dashboard.
    </p>
  </div>
</div>
`;

exports[`OrderDetails should render credit seat message 1`] = `
<div>
  <div
    class="basket-section"
  >
    <h5
      aria-level="2"
    >
      Order Details
    </h5>
    <p>
      After you complete your order you will receive credit for your course.
    </p>
  </div>
</div>
`;

exports[`OrderDetails should render enrollment code product message 1`] = `
<div>
  <div
    class="basket-section"
  >
    <h5
      aria-level="2"
    >
      Order Details
    </h5>
    <p>
      By purchasing, you and your organization agree to the following terms:
    </p>
    <ul>
      <li>
        Each code is valid for the one course covered and can be used only one time.
      </li>
      <li>
        You are responsible for distributing codes to your learners in your organization.
      </li>
      <li>
        Each code will expire in one year from date of purchase or, if earlier, once the course is closed.
      </li>
      <li>
        If a course is not designated as self-paced, you should confirm that a course run is available before expiration.
      </li>
      <li>
        You may not resell codes to third parties.
      </li>
      <li>
        All edX for Business sales are final and not eligible for refunds.
      </li>
    </ul>
    <p>
      You will receive an email at staff@example.com with your enrollment code(s).
    </p>
  </div>
</div>
`;

exports[`OrderDetails should render seat message 1`] = `
<div>
  <div
    class="basket-section"
  >
    <h5
      aria-level="2"
    >
      Order Details
    </h5>
    <p>
      After you complete your order you will be automatically enrolled in the course.
    </p>
  </div>
</div>
`;

exports[`OrderDetails should render verified seat message 1`] = `
<div>
  <div
    class="basket-section"
  >
    <h5
      aria-level="2"
    >
      Order Details
    </h5>
    <p>
      The above total includes any applicable taxes. After you complete your order you will be automatically enrolled in the verified track of the course.
    </p>
  </div>
</div>
`;
